### PR TITLE
[16.07] Additional hardening for older releases

### PR DESCRIPTION
**Pull request by natefoo**

### CODE REVIEW

File: lib/tool_shed/galaxy_install/tool_dependencies/recipe/step_handler.py [hunk 1]
```diff
@@ -165,28 +166,13 @@ def url_download( self, install_dir, downloaded_file_name, download_url, extract
...
-            raise Exception( err_msg )
-        finally:
-            if src:
-                src.close()
-            if dst:
-                dst.close()
+        src = requests.get(download_url, stream=True)
+        with open(file_path, 'wb') as dst:
+            for chunk in src.iter_content(chunk_size=basic_util.CHUNK_SIZE):
+                dst.write(chunk)
```
Comment (nsoranzo) [inline lib/tool_shed/galaxy_install/tool_dependencies/recipe/step_handler.py]: This should be:
```python
if chunk:
    dst.write(chunk)
```
to filter out keep-alive new chunks.

Comment (natefoo) [reply lib/tool_shed/galaxy_install/tool_dependencies/recipe/step_handler.py]: I copied this from @erasche's code in #4604 and still messed it up somehow... thanks as always for catching my mistakes.

**Comment by natefoo** [comment]
The toolshed test failures are legitimate, I'm not sure how we want to fix those. For the purposes of testing it's reasonable to install from a file, but otherwise it shouldn't be possible.

**Comment by jmchilton** [comment]
These recipes install arbitrary software from the Internet - this isn't the avenue of attack one would take if these recipes could be compromised. I'd just remove the "fix" and call it a day.

Update: ... and they are deprecated. We shouldn't be spending time on this.

**Comment by natefoo** [comment]
Ok thanks

**Review by nsoranzo — APPROVED** [review]

**Review by mvdbeek — APPROVED** [review]

**Comment by mvdbeek** [comment]
Thanks @natefoo !

**Pull request merged**